### DISCUSSION TITLE
how to add margin-left?

### DISCUSSION THREAD

**Discussion by igotfr** [KaTeX/KaTeX#2751] [Q&A]
I want that this code:

$`\qquad (\ldots, \quad \allowbreak a_1, \quad \allowbreak \underbrace{a_1 + r}_{\mathnormal{\normalsize{a_2}}}, \quad \allowbreak \underbrace{a_1 + 2r}_{\mathnormal{\normalsize{a_3}}}, \quad \allowbreak \underbrace{a_1 + 3r}_{\mathnormal{\normalsize{a_4}}}, \quad \allowbreak \ldots, \quad \allowbreak \underbrace{a_1 + (n - 3) \cdot r}_{\mathnormal{\normalsize{a_{n-2}}}}, \quad \allowbreak \underbrace{a_1 + (n - 2) \cdot r}_{\mathnormal{\normalsize{a_{n-1}}}}, \quad \allowbreak \underbrace{a_1 + (n - 1) \cdot r}_{\mathnormal{\normalsize{a_{n}}}}, \quad \allowbreak \ldots)`$

give a spacing on automatic break line (\allowbreak), the spacing must be the same as inicial spacing, is there any way?

**Comment by edemaine** [comment]
I don't think this is possible with KaTeX, which doesn't deal with text-level formatting (margin-left).  But it's probably possible in your containing environment, with something like:

```html
<div style="margin-left: 2em">
$(\ldots, \quad \allowbreak a_1, \quad \allowbreak \underbrace{a_1 + r}_{\mathnormal{\normalsize{a_2}}}, \quad \allowbreak \underbrace{a_1 + 2r}_{\mathnormal{\normalsize{a_3}}}, \quad \allowbreak \underbrace{a_1 + 3r}_{\mathnormal{\normalsize{a_4}}}, \quad \allowbreak \ldots, \quad \allowbreak \underbrace{a_1 + (n - 3) \cdot r}_{\mathnormal{\normalsize{a_{n-2}}}}, \quad \allowbreak \underbrace{a_1 + (n - 2) \cdot r}_{\mathnormal{\normalsize{a_{n-1}}}}, \quad \allowbreak \underbrace{a_1 + (n - 1) \cdot r}_{\mathnormal{\normalsize{a_{n}}}}, \quad \allowbreak \ldots)$
</div>
```
